src/main/java/org/kohsuke/github/GHRepositoryStatistics.java

```diff
@@ -0,0 +1,444 @@
+package org.kohsuke.github;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * Statistics for a GitHub repository.
+ *
+ * @author Martin van Zijl
+ */
+public class GHRepositoryStatistics {
+
+    private final GHRepository repo;
+    private final GitHub root;
+
+    private static final int MAX_WAIT_ITERATIONS = 3;
+    private static final int WAIT_SLEEP_INTERVAL = 5000;
+
+    public GHRepositoryStatistics(GHRepository repo) {
+        this.repo = repo;
+        this.root = repo.root;
+    }
+
+    /**
+     * Get contributors list with additions, deletions, and commit count. See
+     * https://developer.github.com/v3/repos/statistics/#get-contributors-list-with-additions-deletions-and-commit-counts
+     */
+    public PagedIterable<ContributorStats> getContributorStats() throws IOException, InterruptedException {
+        return getContributorStats(true);
+    }
+
+    /**
+     * @param waitTillReady Whether to sleep the thread if necessary until the
+     * statistics are ready. This is true by default.
+     */
+    @SuppressWarnings("SleepWhileInLoop")
+    @SuppressFBWarnings(value = {"RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE"}, justification = "JSON API")
+    public PagedIterable<ContributorStats> getContributorStats(boolean waitTillReady) throws IOException, InterruptedException {
```

Comment:
There's a similar need for retry loop with PullRequests and checking for whether they are mergable (and getting `mergeCommitSha`).   Onc we publish this, we are commited to supporting it going forward, but I'm not sure yet that this is model I'd like for follow.  

Would you be willing to mark this as `@Preview @Deprecated`?  That will give the project the freedom to make breaking changes (such as moving this behavior to a common helper the does `pollFor(...)`).

Reply:
Makes sense, I've marked it as suggested.

Should I make the default value `false`, to bring it in line with the other getters?

Reply:
Meh.  Let's leave it.  It is definitely better behavior.